modules/store/src/action_creator.ts

```diff
@@ -124,7 +124,14 @@ export function createAction<T extends string, C extends Creator>(
   }
 }
 
-export function props<P extends object>(): ActionCreatorProps<P> {
+export function props<
+  P extends R extends any
```

Comment:
An afterthought...
Could we extract this into a PropsNotAllowed type?
Similar to the [NotAllowedCheck](https://github.com/ngrx/platform/blob/master/modules/store/src/models.ts#L83) type.